### ISSUE TITLE
Virtual Table does not re-render when the height value changes

### ISSUE THREAD

**Issue by zachrickards** [DevExpress/devextreme-reactive#1744]
I'm using AutoSizer around a virtual table react grid. However, the height changing, which I have verified does not cause a re-render or size change of the grid / virtual table itself.

I have tried using the updated way of the docs, however, that only works with a fixed height on my Paper element surrounding the grid itself.

- [X] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...

- [X] React Grid
- [ ] React Chart
- [ ] React Scheduler

### Current Behaviour

The height value changing on the VirtualTable element doesn't cause a resize of the table itself.

### Expected Behaviour

I expect the height value to be reflect in the UI.

### Steps to Reproduce (for Bugs)

```
   <AutoSizer disableWidth>
     {({ height }) => {
       return (
         <Grid rows={questions} columns={columns} selection={[1]}>
           <SelectionState onSelectionChange={selection => this.changeGridSelection(selection)} />
           <FilteringState defaultFilters={[]} />
           <IntegratedFiltering />
           <SortingState defaultSorting={[]} />
           <IntegratedSorting />
           <VirtualTable columnExtensions={tableColumnExtensions} height={height} />
           <TableHeaderRow showSortingControls />
           <TableFilterRow />
           <TableSelection selectByRowClick highlightSelected showSelectionColumn={false} />
         </Grid>
       );
     }}
  </AutoSizer>
```

### Environment

- devextreme-reactive: 1.9
- react: 16.6.3
- material-ui: 3.5.1

**Comment by ushkal** [comment]
Hi,
 
I couldn't reproduce this behavior. Would you please modify this [sample project](https://codesandbox.io/s/48jn365x8x) to illustrate the problem?

**Comment by hezymal** [comment]
[This](https://codesandbox.io/s/xzv9m7n14q) is debug example. I assume, event "scroll" not emit in "dx-react-core.Sizer" component where stretch HTML-element, but not exactly.

**Comment by patrick-moore** [comment]
I'm also running into the same issue. Here's a gif of the linked sample project:

![resize](https://user-images.githubusercontent.com/26011834/53281673-653d1c00-36f2-11e9-8646-2e00bf2953d9.gif)

**Comment by RopoMen** [comment]
Hi,

I was reading through [@devextreme-reactive CHANGELOG](https://github.com/DevExpress/devextreme-reactive/blob/master/CHANGELOG.md) and saw this [issue as solved](https://github.com/DevExpress/devextreme-reactive/issues/1763) 

I thought that "WOHOOO, they have fixed VirtualTable height issue that I had", tested and noticed that that issue didn't solve my height issue with VirtualTable.

I'm using Bootstrap 4 through [react-bootstrap](https://react-bootstrap.github.io/) I have latest 1.10.3 version of devextreme-reactive components. I render Grid component inside react-bootstrap `Collapse` component. My setup is roughly like this. I'm **only getting 4 rows rendered** even though VirtualTable height is the default is ~500px and I'm expecting about 10 rows to be rendered.

```
<Card>
  <Card.Header>
    <p
      onClick={() => this.props.collapseToggle(collapseId)}
      aria-controls={`${collapseId}`}
      className="mb-0"
      aria-expanded={isOpen}
    >
      Some header
    </p>
  </Card.Header>
  <Collapse in={isOpen}>
    <div>
      <Card.Body id={`${collapseId}`}>
        <Grid rows={this.props.items} columns={this.props.columns}>
          <SelectionState
            onSelectionChange={this.onSelectionChanged}
            selection={this.selectedIndexes}
          />
          <IntegratedSelection />

          <VirtualTable
            tableComponent={props => (
              <Table.Table
                {...props}
                className="table-sm table-borderless"
              />
            )}
            estimatedRowHeight="32"
          />
          <TableSelection
            selectByRowClick
            cellComponent={({ row, selected, onToggle }) => {
              return (
                <td>
                  <Form.Group
                    className="m-0"
                    controlId={`${row.label}-${row.id}`}
                  >
                    <Form.Check
                      type="checkbox"
                      label="&nbsp;"
                      custom
                      checked={selected}
                      onChange={onToggle}
                    />
                  </Form.Group>
                </td>
              );
            }}
          />
        </Grid>
      </Card.Body>
    </div>
  </Collapse>
</Card>
```

**Comment by ushkal** [comment]
Hi @RopoMen 

I've tried to reproduce the issue but all rows are rendered as expected. Would you please modify this Codesandbox to demonstrate the problem - https://codesandbox.io/s/z3wy9nvyk4?

**Comment by DanHarman** [comment]
Repro steps of my issue:

- Use existing example from @patrick-moore 
- Zoom out in browser using ctrl-mousewheel.
- Hit refresh button on the codesandbox virtual browser.
- Marvel at only seeing 4 rows of data.
- Reset zoom to 100%.
- Hit refresh button on virtual browser.
- Everything ok again.

**Comment by DanHarman** [comment]
@ushkal Would it be better if I opened a separate ticket for this? It's a pretty big problem and blocking me from doing a release. I don't really want to have to revert many versions to have a working version though as then it impacts plugins I've written etc.

**Comment by RopoMen** [comment]
@ushkal 
I tried to re-factor that example close as possible to what I have from layout perspective, but I was unable to reproduce that 4 row rendering issue with that, my modifications are here https://codesandbox.io/s/v3x712yy3

My setup contains also redux + reselect and items are coming through that way inside the component. But that should not affect on rendering, only thing that came in my mind is that perhaps "connect()" from react-redux has tiny tiny delay that would case the issue.

I also ensured that this is not browser specific issue by testing my app with Chrome and Firefox.
